### PR TITLE
Delete initial Multiapps after initial execution

### PR DESCRIPTION
**Pull request by permcody**
We now delete multiapps and transfers that are only needed for initial execute. This case arises with `FullSolveMultiApp` in Rattlesnake quite often. This capability is being added to reduce memory overhead.

### CODE REVIEW

**Comment by YaqiWang** [comment]
Other Yak failing tests other than recover tests can probably fixed easily by @snschune .

File: framework/src/base/FEProblemBase.C [hunk 1]
```diff
@@ -672,6 +672,14 @@ void FEProblemBase::initialSetup()
     for (THREAD_ID tid = 0; tid < n_threads; ++tid)
         _assembly[tid]->initNonlocalCoupling();
   }
+
+  // Remove some of the heavier objects that are only needed during initial
+  // Several other types of objects may be added here in the future.
+  _multi_apps.clear(EXEC_INITIAL);
```
Comment (YaqiWang) [inline framework/src/base/FEProblemBase.C]: Is this going to delete multapps which are on for example `initial timestep_end`?

**Comment by YaqiWang** [comment]
The initial plan is to shuffle the execution and transfer of multiapps so that an individual multiapp can be deleted once it finishes its job, while other multiapps may still not be scheduled.

File: framework/src/base/FEProblemBase.C (hunk 1 above)
Comment (permcody) [inline framework/src/base/FEProblemBase.C]: No - I put that in the documentation and provided a test case to show that it works properly.

**Comment by snschune** [comment]
Looking at this one first, then #8467.

**Comment by snschune** [comment]
@permcody Just to make sure I understand the setup correctly: there is a Warehouse for each exec option?

From what I've seen is that most yak failures have Multiapps on `custom` which are in the execute on initial warehouse.

**Comment by permcody** [comment]
Hmm... There should be a separate warehouse for each exec option... Might have to dig into that a bit.

**Comment by snschune** [comment]
And the other test that fails, is because it wants to write a checkpoint file & back up the multiapp after it has been deleted.

**Comment by permcody** [comment]
OK - well I'll have to look at that too.

**Comment by permcody** [comment]
Ugh, this is getting harder still. The `RestartableData` objects still have pointers to the deleted objects so that's the new challenge. I'm looking into it to see if we can get around that.

**Comment by permcody** [comment]
@YaqiWang - Perhaps we should try this on your real problems and see if it's worth pursuing. I removed the magic line of code to actually delete because it causes a few tests to fail. It's definitely deleting the code but I think it might be quite the rabbit hole to figure out why it's not working in all cases.

**Comment by YaqiWang** [comment]
I am swamped as well. But I do want this capability and this can improve our multiapp implementation. It can help us think of the lifetime of a multiapp and the proper way of handling recovery as well imo. I can suggest that we close this PR, but leave the issue open. The way this PR is approaching is not our initial plan (shuffling the maltiapp execution and transfers) anyway.

**Comment by YaqiWang** [comment]
BTW, this is no longer a blocking issue to us because we can use variable restart capability to do this by assuming the same mesh is used.

**Comment by permcody** [comment]
Closing, I'll revive this later if somebody wants it.

**Pull request closed without merging**